### ISSUE TITLE
Configuring the Swagger output for Swagger-UI

### ISSUE THREAD

**Issue by ncreep** [daviddenton/fintrospect#39]
Hi,

I'm trying to use the Swagger output from a route-module that I've defined in Swagger-UI and I've stumble upon a number of issues.

With Swagger-UI 3.0.x I'm getting:
```
Errors
Parser error 
duplicated mapping key
```
Not sure what's at fault here, but after downgrading to 2.2.x, I did manage to parse the JSON and render it in the UI. 

Now, to be able to hit the "Try it out!" button in the UI I need to specify the `host` and `schemes` keys in the generated Swagger JSON.
I've quickly hacked-up a modified `Swagger2dot0Json` class to allow these modifications, see [this gist](https://gist.github.com/ncreep/7c87c61ee62bd0760c8828c93a451f02), and it seems to work.

For a proper change I'm guessing that passing the relevant info in the some dedicated class would be more appropriate.

If you find any of this useful, I can open a pull request.

Thanks.

**Comment by ncreep** [comment]
I now managed to find the reason why loading failed in Swagger-UI 3.0.x, it's because I've used the same status code twice in a `returning` call on `RouteSpec`.
Though it may be beneficial to either validate that there is only possible response per response code (or at least adding a comment about it in the API). If someone can point me where to do that, I can make a pull request.

Now it loads in the latest UI, and the problem with `host`/`schemes` is no longer relevant.

**Comment by daviddenton** [comment]
It's a good point about the Response validation. I've reopen this and we can look at fixing that.

I also suspect there may be a problem with Swagger v3 UI after all as I'm also getting the same error on one of my apps.

**Comment by ncreep** [comment]
Regarding the v3 UI, the error I had was coming from a duplicate key from a JSON along the line of:
```json
{
  "200": "...",
  "400": "...",
  "200": "..."
}
```
Swagger is trying to parse it as YAML map and it fails on a duplicate key.

So make sure that you don't generate JSON "maps" with duplicate keys anywhere.

**Comment by daviddenton** [comment]
Cool. Thanks for the clarification.  :)

**Comment by daviddenton** [comment]
This was indeed 2 problems with duplication. Both are fixed in #14.17.3.

Duplicate responses: For the moment, have simply converted the Sequence of responses into a Map. This means that we don't get a runtime error on rendering, but neither do we get a runtime error when adding the response. However, this would allow the "overriding" of response codes in situations where the route specs are reused. Possibly look at this again if that turns out to be a mistake :)

Swagger rendering: The renderer was producing duplicate object definitions when a single type of object was returned in multiple responses (or consumed bodies).
